### PR TITLE
PLT-1883 Update search result components when we receive channels

### PR DESCRIPTION
**Pull request by hmhealey**
The reason that the channel names in the search results don't show up sometimes is because the channels haven't been retrieved yet and then once they are retrieved, the search results aren't updated.

I also increased the maximum number of listeners on the ChannelStore by 100 since there will only be 100 search results returned except in incredibly rare circumstances. Any added by this will be very short lived anyway

### CODE REVIEW

**Comment by jwilander** [comment]
I'd much rather the parent of search_results_item have the listener and then pass the channel in as a prop to search_results_item. That way we don't have to raise the maximum number of listeners to 100

**Comment by jwilander** [comment]
Especially since we'd have to change the maximum number of listeners every time we increase the number of results returned

**Comment by hmhealey** [comment]
I'll see if that works without rerendering the entire search result list every time the channel store changes. I was doing it in the items since they knew whether or not their individual channels existed so it would only need to bind listeners when necessary.

**Comment by jwilander** [comment]
Don't worry about re-rendering too much. React is really good at that and it will rarely cause a noticable slowdown. Besides, if the channel doesn't change in the props then the search result won't re-render anyways

**Comment by hmhealey** [comment]
Updated the SearchResults component to store the channels and to just attach a single listener instead of having each item do it separately

**Comment by jwilander** [comment]
+1

File: web/react/components/search_results.jsx [hunk 1]
```diff
@@ -63,6 +83,16 @@ export default class SearchResults extends React.Component {
         }
     }
 
+    handleChannelChange() {
+        const channels = this.getChannelStateFromStores(this.state.results);
+
+        if (!Utils.areMapsEqual(channels, this.state.channels)) {
```
Comment (crspeller) [inline web/react/components/search_results.jsx]: This should be in shouldComponentUpdate().

**Comment by crspeller** [comment]
+1

**Comment by coreyhulen** [comment]
Merging since the jira tix is marked 1.5.

**Pull request merged**